core/modules/wbase/Task.h

```diff
@@ -68,19 +74,38 @@ class TaskScheduler {
     virtual void taskCancelled(Task*)=0;///< Repeated calls must be harmless.
 };
 
+/// Used to find tasks that are in process for debugging with Task::tSeq.
+/// This is largely meant to track down incomplete tasks in a possible intermittent
+/// failure and should probably be removed when it is no longer needed.
+/// It depends on code in BlendScheduler to work. If the decision is made to keep it
+/// forever, dependency on BlendScheduler needs to be re-worked.
+struct IdSet {
+    void add(int id) { std::lock_guard<std::mutex> lock(mx); _ids.insert(id); }
+    void remove(int id) { std::lock_guard<std::mutex> lock(mx); _ids.erase(id); }
+    std::string toString() {
+        std::ostringstream os;
+        std::lock_guard<std::mutex> lock(mx);
+        os << "count=" << _ids.size() << " ";
+        for(auto j: _ids) {
+            os << j << ", ";
+        }
+        return os.str();
+    }
+private:
+    std::set<int> _ids;
+    std::mutex mx;
+};
+
 /// struct Task defines a query task to be done, containing a TaskMsg
 /// (over-the-wire) additional concrete info related to physical
 /// execution conditions.
 /// Task is non-copyable
 /// Task encapsulates nearly zero logic, aside from:
 /// * constructors
-struct Task {
+struct Task : public util::Command {
```

Comment:
Why "struct" here, and not "class"?  Access is immediately overridden to "public" anyway, and there are private members as well.